src/main/java/edu/usc/ir/geo/gazetteer/GeoNameResolver.java

```diff
@@ -99,45 +115,56 @@
 		if (locationNameEntities.size() >= 200)
 			hitsPerPage = 5; // avoid heavy computation
 		IndexSearcher searcher = new IndexSearcher(reader);
-
+		
 		Query q = null;
 
-		HashMap<String, ArrayList<ArrayList<String>>> allCandidates = new HashMap<String, ArrayList<ArrayList<String>>>();
+		HashMap<String, List<List<String>>> allCandidates = new HashMap<String, List<List<String>>>();
```

Comment:
@chrismattmann 
As a good practice I always try to keep reference as generic interface freeing it from underlying implementation unless absolutely required. 

I observed this while I was running searchGeoName() through my test code so I changed the references to generic List as a goodwill gesture. Do you think otherwise?